dll/win32/browseui/shellbrowser.cpp

```diff
@@ -3656,24 +3658,90 @@ LRESULT CShellBrowser::OnBackspace(WORD wNotifyCode, WORD wID, HWND hWndCtl, BOO
     return 0;
 }
 
-LRESULT CShellBrowser::OnOrganizeFavorites(WORD wNotifyCode, WORD wID, HWND hWndCtl, BOOL &bHandled)
+static BOOL
+CreateShortcut(
+    IN LPCWSTR pszLnkFileName,
+    IN LPCITEMIDLIST pidl,
+    IN LPCWSTR pszDescription OPTIONAL)
 {
-    CComPtr<IShellFolder> psfDesktop;
-    LPITEMIDLIST pidlFavs;
-    HRESULT hr;
-    hr = SHGetSpecialFolderLocation(m_hWnd, CSIDL_FAVORITES, &pidlFavs);
+    IPersistFile *pPF;
+    IShellLinkW *pSL;
+    HRESULT hr = CoInitialize(NULL);
     if (FAILED(hr))
+        return hr;
+
+    hr = CoCreateInstance(CLSID_ShellLink, NULL, CLSCTX_INPROC_SERVER,
+                          IID_IShellLinkW, (LPVOID*)&pSL);
+    if (SUCCEEDED(hr))
     {
-        hr = SHGetSpecialFolderLocation(m_hWnd, CSIDL_COMMON_FAVORITES, &pidlFavs);
-        if (FAILED(hr))
-            return 0;
+        pSL->SetIDList(pidl);
+
+        if (pszDescription)
+            pSL->SetDescription(pszDescription);
+
+        hr = pSL->QueryInterface(IID_IPersistFile, (LPVOID*)&pPF);
+        if (SUCCEEDED(hr))
+        {
+            hr = pPF->Save(pszLnkFileName, TRUE);
+            pPF->Release();
+        }
+        pSL->Release();
     }
 
+    CoUninitialize();
+
+    return SUCCEEDED(hr);
+}
+
+HRESULT GetFavsLocation(HWND hWnd, LPITEMIDLIST *pPidl)
+{
+    HRESULT hr = SHGetSpecialFolderLocation(hWnd, CSIDL_FAVORITES, pPidl);
+    if (FAILED(hr))
+        hr = SHGetSpecialFolderLocation(hWnd, CSIDL_COMMON_FAVORITES, pPidl);
+
+    return hr;
+}
+
+LRESULT CShellBrowser::OnAddToFavorites(WORD wNotifyCode, WORD wID, HWND hWndCtl, BOOL &bHandled)
+{
+    LPITEMIDLIST pidlFavs;
+    HRESULT hr = GetFavsLocation(m_hWnd, &pidlFavs);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return 0;
+
+    SHFILEINFOW fileInfo = { NULL };
+    if (!SHGetFileInfoW((LPCWSTR)fCurrentDirectoryPIDL, 0, &fileInfo, sizeof(fileInfo),
+                        SHGFI_PIDL | SHGFI_DISPLAYNAME))
+    {
+        return 0;
+    }
+
+    WCHAR szPath[MAX_PATH];
+    SHGetPathFromIDListW(pidlFavs, szPath);
+    PathAppendW(szPath, fileInfo.szDisplayName);
+    PathAddExtensionW(szPath, L".LNK");
```

Comment:
```suggestion
    PathAddExtensionW(szPath, L".lnk");
```
nitpick: Windows seems to use lower-case lnk. Aside from that lgtm.
![image](https://github.com/reactos/reactos/assets/33393466/00d3f18b-6efd-4952-b591-65b839d436ca)